File: api/src/main/java/dev/ulloasp/mlsuite/schema/application/dto/SchemaDraftMergeRequest.java
package dev.ulloasp.mlsuite.schema.application.dto;

import java.util.List;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.NotBlank;

public record SchemaDraftMergeRequest(
        @NotNull Long expectedCurrentVersionId,
        @NotBlank String expectedCurrentDocumentHash,
        @NotNull Long expectedDraftRevision,
        @Valid @NotNull List<@NotNull @Valid SchemaDraftMergeResolutionDto> resolutions) {
}

File: frontend/src/api/schemas/dtos/schema-draft-merge-request.ts
/*
SPDX-License-Identifier: MIT
Copyright (c) 2025 Pablo Ulloa Santin
*/

import type { SchemaDraftMergeResolutionDto } from "./schema-draft-merge-resolution-dto";

export type SchemaDraftMergeRequest = {
  expectedCurrentVersionId: string | number;
  expectedCurrentDocumentHash: string;
  expectedDraftRevision: number;
  resolutions: SchemaDraftMergeResolutionDto[];
};
